programs/lz4cli.c

```diff
@@ -705,7 +708,34 @@ int main(int argc, const char** argv)
             operationResult = LZ4IO_decompressMultipleFilenames(prefs, inFileNames, ifnIdx, !strcmp(output_filename,stdoutmark) ? stdoutmark : LZ4_EXTENSION);
         else
             operationResult = DEFAULT_DECOMPRESSOR(prefs, input_filename, output_filename);
-    } else {   /* compression is default action */
+    } else if (mode == om_list){
+        LZ4F_compFileInfo_t cfinfo;
```

Comment:
In term of design :
try to have as little work as possible within `lz4cli.c`.

`lz4cli.c` primary role is to be a command line parser, and a job dispatcher.

I believe most of this section could probably become a function within `lz4io.c`.